0015-ilp-addresses/0015-ilp-addresses.md

```diff
@@ -0,0 +1,123 @@
+- [Payment rail](#payment-rail)
+- [Ledger prefix](#ledger-prefix)
+- [Subledger and account group](#subledger-and-account-group)
+- [Account identifier](#account-identifier)
+- [Interactions](#interactions)
+
+
+### Neighborhoods
+
+Neighborhoods have no specific meaning, but serve as a quick shortcut for routing to the right area.
+
+- For regional currencies, use a continent code from the folowing table:
+    | Segment | Continent     |
+    |:--------|:--------------|
+    | `af`    | Africa        |
+    | `as`    | Asia          |
+    | `eu`    | Europe        |
+    | `na`    | North America |
+    | `sa`    | South America |
+    | `oc`    | Oceania       |
+    | `an`    | Antarctica    |
+- For decentralized crypto-currency ledgers, use the `crypto` neighborhood.
+- For naturally-ocurring commodities or other globally-distributed currencies, use the `earth` neighborhood.
+- For currencies strongly associated with a particular international corporation, use the continent of that corporation's headquarters.
+
+It may also be useful to have 1-3 levels of "sub-neighborhoods" to group together related connectors or ledgers. We have no specific recommendations at this time.
+
+### Payment Rail
+
+The payment rail refers to a specific way some accounts or ledgers are connected outside of ILP, such as ACH in the U.S., SEPA in Europe, or other online payment systems such as PayPal or Alipay. We recommend using an all-lowercase abbreviation for the payment rail if one is applicable to the address. Examples:
+
+| Segment | Payment Rail                                                       |
+|:--------|:-------------------------------------------------------------------|
+| `ach`   | [United States Automated Clearing House](https://en.wikipedia.org/wiki/Automated_Clearing_House) |
+| `sepa`  | [Single Euro Payments Area](https://en.wikipedia.org/wiki/Single_Euro_Payments_Area) |
+
+***TODO: more rails?***
+
+### Ledger Prefix
+
+The ledger prefix is the unique name of the ledger. When doing a local delivery, the connector typically uses this segment to figure out which ledger plugin to use. As a result, it's difficult (but possible) to make a connector between two ledgers that use the same ledger prefix.
+
+### Subledger and Account Group
+
+Zero or more segments representing optional divisions within a ledger. If there are no fees involved in transferring between a ledger and a subledger, the subledger can be a "local delivery" from the connector's perspective. If fees are necessary for connecting to a subledger, it may require a connector. ***TODO: More clarifications and detailed recommendations needed.***
+
+### Account Identifier
+
+A unique identifier of the account within the ledger. The ledger plugin maps these to accounts within a ledger. For some ledgers, a simple conversion rule may suffice; other ledgers may require a lookup table. ***TODO: clarify exactly how five-bells-ledger-plugin or Common Ledger API plugin do this***
```

Comment:
Very interesting question. Address translation should definitely be up to the ledger plugin imo. Whoever creates the ledger plugin will have a much better idea what would make sense for them than we do.

This is also related to the distinction I made between subledger and account group.

A subledger is a different system (think separate physical servers) whereas an account group is just a subdivision or grouping of a accounts within one ledger (think single database but accounts are grouped together).

To enable account groups I think the ledger would indeed strip off multiple segments and I could see that being useful.

Imagine an accounting ledger which uses `assets` and `liabilities` as account groups. So you could have an address like `us.example.assets.1012` and an address like `us.example.liabilities.7209` and they'd be translated into the local identifiers `assets:1012` and `liabilities:7209`, respectively. (Notice that I used a colon, which might just happen to be the format that this hypothetical accounting ledger uses for its local identifiers.) Nothing seems particularly wrong to me in this picture.